src/client/Microsoft.Identity.Client/AppConfig/CiamAuthorityHelper.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client.AppConfig
+{
+    /// <summary>
+    /// Translates CIAM authorities into a usable form. This is needed only until ESTS is updated to support the north star format
+    /// North star format: https://idgciamdemo.ciamlogin.com
+    /// Transformed format: https://idgciamdemo.ciamlogin.com/idgciamdemo.onmicrosoft.com
+    /// </summary>
+    internal class CiamAuthorityHelper
```

Comment:
Can the logic in this class be moved to CiamAuthority class, better enacapsulation?

Reply:
The problem I ran into was the need to have the transformed authority in the `AuthorityInfo` class, which is needed for the constructor of the `CiamAuthority`. So you would end up with a catch 22 of sorts.

Reply:
Ah yea, maybe just make it a static helper method - so it would be CiamAuthority.TransformAuthority or something.